enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationPrecompleteGuard.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * PowerAuth Enrollment Server
+ * Copyright (C) 2022 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.app.onboardingserver.impl.service;
+
+import com.wultra.app.enrollmentserver.model.enumeration.*;
+import com.wultra.app.onboardingserver.common.database.OnboardingOtpRepository;
+import com.wultra.app.onboardingserver.common.database.entity.DocumentVerificationEntity;
+import com.wultra.app.onboardingserver.common.database.entity.IdentityVerificationEntity;
+import com.wultra.app.onboardingserver.common.database.entity.OnboardingOtpEntity;
+import com.wultra.app.onboardingserver.common.errorhandling.RemoteCommunicationException;
+import com.wultra.app.onboardingserver.configuration.IdentityVerificationConfig;
+import com.wultra.app.onboardingserver.statemachine.guard.document.RequiredDocumentTypesGuard;
+import com.wultra.security.powerauth.client.v3.ActivationStatus;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase.*;
+import static com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.ACCEPTED;
+import static com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.VERIFICATION_PENDING;
+
+/**
+ * Validate all critical conditions were met before finishing the onboarding.
+ * <p>
+ * This should never happen for the state machine.
+ * It works as a safety stop.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+// TODO (racansky, 2022-10-14) consider make it Guard for Spring State Machine
+@Component
+@Slf4j
+class IdentityVerificationPrecompleteGuard {
```

Comment:
I am not sure we should call this class a `guard` when it is not an actual guard for state machine.

Reply:
I agree that it may be confusing a little bit but the state machine is not the only one using the keyword `guard`.  IMHO `validator` is something much simpler. What about `check`? There is the same issue with `RequiredDocumentTypesGuard`.

Reply:
`Check` or `Validator` is fine, let's just not confuse reader with state machine terminology.